apollo-federation/src/query_graph/graph_path.rs

```diff
@@ -71,14 +72,23 @@ use crate::schema::ValidFederationSchema;
 
 use super::condition_resolver::ContextMapEntry;
 
-#[derive(Clone, serde::Serialize)]
+#[derive(Clone, serde::Serialize, Debug, Eq)]
 pub(crate) struct ContextAtUsageEntry {
     pub(crate) context_id: String,
     pub(crate) relative_path: Vec<String>,
     pub(crate) selection_set: SelectionSet,
     pub(crate) subgraph_arg_type: Type,
 }
 
+impl PartialEq for ContextAtUsageEntry {
+    fn eq(&self, other: &Self) -> bool {
+        self.context_id == other.context_id
+            && self.relative_path == other.relative_path
+            && self.selection_set == other.selection_set
+            && self.subgraph_arg_type == other.subgraph_arg_type
+    }
+}
```

Comment:
`PartialEq` can be derived.
```suggestion
#[derive(Clone, serde::Serialize, Debug, Eq)]
pub(crate) struct ContextAtUsageEntry {
    pub(crate) context_id: String,
    pub(crate) relative_path: Vec<String>,
    pub(crate) selection_set: SelectionSet,
    pub(crate) subgraph_arg_type: Type,
}
```